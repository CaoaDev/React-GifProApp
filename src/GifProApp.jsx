import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifProApp = ( {defaultAnime= [] } ) => {
    // const animes = ['One Punch Man', 'Samurai X', 'Dragon Ball'];
    // const [ anime, setAnime ] = useState (['']);
    
    const [ anime, setAnime ] = useState ( defaultAnime );

    // const handleAdd = () => {
    //     setAnime( anime => [...anime, 'HunterxHunter', 'Madoka Magica', 'Kimetsu'] );
    // }
    // para agregar una lista a setanimes

    return (
        <>
        <h2><center>GifProApp</center></h2> 
        <AddCategory setAnime={ setAnime } />
        <hr/>
    
        <h1>
            <ol>
                {
                    anime.map( anime =>(
                         <GifGrid 
                         key={ anime }
                         anime ={ anime } />
                         ))
                }
            </ol>
        </h1>
        </>
    )
}
