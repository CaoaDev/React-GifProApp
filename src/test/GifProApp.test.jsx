import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifProApp } from '../GifProApp';

describe('Mostrar correctamente GifProApp', () => {
    
    const anime= 'kimetsu';
    const wrapper= shallow( <GifProApp anime={ anime }  /> );

    test('Debe mandar el snapshot correctmanete', () => {
      
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de mostrar una lista de anime', () => {
        
        const anime= ['kimetsu', 'naruto'];
        const wrapper= shallow( <GifProApp defaultAnime={ anime }  /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( anime.length );

    });
    

});
