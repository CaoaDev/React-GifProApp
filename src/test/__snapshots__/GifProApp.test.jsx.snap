// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mostrar correctamente GifProApp Debe de mostrar una lista de anime 1`] = `
<Fragment>
  <h2>
    <center>
      GifProApp
    </center>
  </h2>
  <AddCategory
    setAnime={[Function]}
  />
  <hr />
  <h1>
    <ol>
      <GifGrid
        anime="kimetsu"
        key="kimetsu"
      />
      <GifGrid
        anime="naruto"
        key="naruto"
      />
    </ol>
  </h1>
</Fragment>
`;

exports[`Mostrar correctamente GifProApp Debe mandar el snapshot correctmanete 1`] = `
<Fragment>
  <h2>
    <center>
      GifProApp
    </center>
  </h2>
  <AddCategory
    setAnime={[Function]}
  />
  <hr />
  <h1>
    <ol />
  </h1>
</Fragment>
`;
